src/url_aggregator.cpp

```diff
@@ -459,19 +459,79 @@ bool url_aggregator::set_hostname(const std::string_view input) {
 }
 
 [[nodiscard]] std::string_view url_aggregator::get_username() const noexcept {
+  /**
+   * The username, if it exists, starts at protocol_end+3 (inclusively) and
+   * ends at username_end (non-included). Yet the code below does not seem
+   * to agree?
+   */
+  /**
+   * https://user@pass:example.com:1234/foo/bar?baz#quux
+   *      |      |    |          | ^^^^|       |   |
+   *      |      |    |          | |   |       |   `----- hash_start
+   *      |      |    |          | |   |       `--------- search_start
+   *      |      |    |          | |   `----------------- pathname_start
+   *      |      |    |          | `--------------------- port
+   *      |      |    |          `----------------------- host_end
+   *      |      |    `---------------------------------- host_start
+   *      |      `--------------------------------------- username_end
+   *      `---------------------------------------------- protocol_end
+   */
   if (has_authority() && components.username_end > components.protocol_end + 2) {
     return helpers::substring(buffer, components.protocol_end + 2, components.username_end);
   }
   return "";
 }
 
 [[nodiscard]] std::string_view url_aggregator::get_password() const noexcept {
+  /**
+   * The password, if it exists, starts at username_end+1 (inclusively) and
+   * ends at host_start (non-included).
+   */
+  /**
+   * https://user@pass:example.com:1234/foo/bar?baz#quux
```

Comment:
This is wrong. I looked and the original `url_components.h` is also wrong.

```suggestion
   * https://user:pass@example.com:1234/foo/bar?baz#quux
```